encode.go

```diff
@@ -233,10 +236,23 @@ func isBase60Float(s string) (result bool) {
 	return base60float.MatchString(s)
 }
 
+func isDate(s string) (result bool) {
+	return date.MatchString(s)
+}
+
 // From http://yaml.org/type/float.html, except the regular expression there
 // is bogus. In practice parsers do not enforce the "\.[0-9_]*" suffix.
 var base60float = regexp.MustCompile(`^[-+]?[0-9][0-9_]*(?::[0-5]?[0-9])+(?:\.[0-9_]*)?$`)
 
+// From http://yaml.org/type/timestamp.html
+var date = regexp.MustCompile(`[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]|[0-9][0-9][0-9][0-9]-[0-9][0-9]?-[0-9][0-9]?([Tt]|[ \t]+)[0-9][0-9]?:[0-9][0-9]:[0-9][0-9](\.[0-9]*)?(([ \t]*)Z|[-+][0-9][0-9]?(:[0-9][0-9])?)?`)
+
+func (e *encoder) timev(tag string, in reflect.Value) {
+	m := in.Interface().(encoding.TextMarshaler)
```

Comment:
I don't handle the case of in.Interface() not being a TextMarshaler because that's checked earlier on. I guess in doesn't need to be reflect.Value, but I left it that way because the rest of the functions are also like that.